mllib/src/test/scala/org/apache/spark/ml/FunctionsSuite.scala

```diff
@@ -101,4 +102,41 @@ class FunctionsSuite extends MLTest {
     val resultVec3 = df3.select(array_to_vector(col("c1"))).collect()(0)(0).asInstanceOf[Vector]
     assert(resultVec3 === Vectors.dense(Array(1.0, 2.0)))
   }
+
+  test("SPARK-45859: 'functions$' should not be affected by a broken class loader") {
+    quietly {
+      // Only one SparkContext should be running in this JVM (see SPARK-2243)
+      sc.stop()
```

Comment:
shouldn't we add this after each test run ?

Reply:
good point, we'd better close `sc` in this UT, in case contributors add new UTs after this one and tested with non-expected `sc`